src/module.c

```diff
@@ -578,9 +578,6 @@ static int internalAdd(RedisModuleCtx *ctx, Series *series, api_timestamp_t time
     if (retval == TSDB_ERR_TIMESTAMP_TOO_OLD) {
         RedisModule_ReplyWithError(ctx, "TSDB: Timestamp cannot be older than the latest timestamp in the time series");
         result = REDISMODULE_ERR;
-    } else if (retval != TSDB_OK) {
```

Comment:
@ashtul please revert this, its a defensive code against unknown error, its better to leave it there in case that we have an unexpected error.

Reply:
@danni-m , there is no `unknown error`. We have only two possible return values, `TSDB_ERR_TIMESTAMP_TOO_OLD` and `TSDB_OK`.

Reply:
There is no other errors as of now, correct. but if its not `TSDB_OK` we should be defensive and return an error.

Reply:
This, by definition, makes it `dead code`.
It cannot get coverage.

Reply:
There are 2 options:
1. remove this. 
Pros: we will have better coverage because we don't hit this branch when testing.
Cons: we don't check that the return value is OK, and we might hit some other error and we will continue as if it pased.
2.  keep this:
Pros: we always check that the response is OK and we will continue only if it was succesfull
Cons: this branch is not covered in testing.

The risk of #1 is way way bigger than #2.
Regarding code coverage, its just a metric to give you a feel of how much of the code is being tested, its not an absolute metric or a goal by itself.

Reply:
agree to disagree and return handling of != OK